src/login/logind-user.c

```diff
@@ -355,6 +355,100 @@ static void user_start_service(User *u) {
                                "Failed to start user service '%s', ignoring: %s", u->service, bus_error_message(&error, r));
 }
 
+static int update_slice_callback(sd_bus_message *m, void *userdata, sd_bus_error *ret_error) {
+        _cleanup_(user_record_unrefp) UserRecord *ur = userdata;
+
+        assert(m);
+        assert(ur);
+
+        if (sd_bus_message_is_method_error(m, NULL)) {
+                log_error_errno(sd_bus_message_get_errno(m),
+                                "Failed to update slice of %s: %s",
+                                ur->user_name,
+                                sd_bus_message_get_error(m)->message);
+
+                return 0;
+        }
+
+        log_debug("Successfully set slice parameters of %s.", ur->user_name);
+        return 0;
+}
+
+static int user_update_slice(User *u) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
+        int r;
+
+        assert(u);
+
+        if (u->user_record->tasks_max == UINT64_MAX &&
+            u->user_record->memory_high == UINT64_MAX &&
+            u->user_record->memory_max == UINT64_MAX &&
+            u->user_record->cpu_weight == UINT64_MAX &&
+            u->user_record->io_weight == UINT64_MAX)
+                return 0;
+
+        r = sd_bus_message_new_method_call(
+                        u->manager->bus,
+                        &m,
+                        "org.freedesktop.systemd1",
+                        "/org/freedesktop/systemd1",
+                        "org.freedesktop.systemd1.Manager",
+                        "SetUnitProperties");
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        r = sd_bus_message_append(m, "sb", u->slice, true);
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        r = sd_bus_message_open_container(m, 'a', "(sv)");
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        if (u->user_record->tasks_max != UINT64_MAX) {
+                r = sd_bus_message_append(m, "(sv)", "TasksMax", "t", u->user_record->tasks_max);
+                if (r < 0)
+                        return bus_log_create_error(r);
+        }
+
+        if (u->user_record->memory_high != UINT64_MAX) {
```

Comment:
C&P here.

I think it'd be better to make this into a loop. The type is `t` for all of them, so the code wouldn't' be complicated.

Reply:
hm, how would that look like?